client/lib/user-settings/index.js

```diff
@@ -60,6 +61,11 @@ UserSettings.prototype.fetchSettings = function() {
 	debug( 'Fetching user settings' );
 	wpcom.me().settings( function( error, data ) {
 		if ( ! error ) {
+			traverseAndDecodeEntities( data, [
+				'display_name',
+				'description',
+				'user_URL'
+			] );
```

Comment:
I agree that this is a good place to escape the values. If you used a higher-order function that returned the mapped values, you could even do something like: `this.settings = traverseAndDecode( data );`